### PR TITLE
Splitting of GRP object 

### PR DESCRIPTION
**Pull request by chiarazampolli**
Object to fill information from ECS, see https://alice.its.cern.ch/jira/browse/OCTRL-564

### CODE REVIEW

File: DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  /// print itself
+  void print() const;
+
+  static GRPECSObject* loadFrom(const std::string& grpecsFileName = "", const std::string& grpecsName = "GRPECS");
+
+ private:
+  timePoint mTimeStart = 0; ///< DAQ_time_start entry from DAQ logbook
+
+  uint32_t mNHBFPerTF = 256; /// Number of HBFrames per TF
```
Comment (shahor02) [inline DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h]: I think we can set the default to 128

File: macro/CreateGRPECSObject.C [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  grpecs.setDetsTrigger(detsTrigger);
+  grpecs.setRun(run);
+  grpecs.setDataPeriod(dataPeriod);
+
+  CcdbApi api;
+  api.init(ccdbPath);
+  std::map<std::string, std::string> metadata;
+  metadata["responsible"] = "ECS";
+  long ts = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+  api.storeAsTFileAny(&grpecs, "GRP/GRP/ECS", metadata, ts, ts + 60 * 60 * 24 * 365); // making it 1-year valid to be sure we have something
```
Comment (shahor02) [inline macro/CreateGRPECSObject.C]: I think the start of validity should be equal to `start` time (but in `ms`), I would also provide an end of validity, with the convention that <0 is interpreted as start + e.g. 10h, to cover the fill.

File: macro/CreateGRPECSObject.C [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  grpecs.setDetsReadOut(detsReadout);
+  grpecs.setDetsContinuousReadOut(detsContinuousRO);
+  grpecs.setDetsTrigger(detsTrigger);
+  grpecs.setRun(run);
+  grpecs.setDataPeriod(dataPeriod);
+
+  CcdbApi api;
+  api.init(ccdbPath);
+  std::map<std::string, std::string> metadata;
+  metadata["responsible"] = "ECS";
```
Comment (shahor02) [inline macro/CreateGRPECSObject.C:58]: For the run-specific objects we agreed to pass metadata runNumber

**Review by shahor02 — COMMENTED** [review]
Thanks @chiarazampolli ! 
Please see a few comments below

File: macro/CreateGRPECSObject.C (hunk 2 above)
Comment (shahor02) [inline macro/CreateGRPECSObject.C]: Also, I would store it in e.g. `GLO/Config/GRPECS`, we anyway have plenty of other global objects, better collect them in one place.

File: DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h (hunk 1 above)
Comment (chiarazampolli) [reply DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h]: done

File: macro/CreateGRPECSObject.C (hunk 3 above)
Comment (chiarazampolli) [reply macro/CreateGRPECSObject.C:58]: done

**Comment by chiarazampolli** [comment]
Hi @shahor02 , 

I moved this PR to WIP, I would add here the split to the MagField GRP object and some changes in the LHCIF object, to avoid too many PRs. 

Chiara

**Comment by chiarazampolli** [comment]
Hello @shahor02 . 
I implemented the filling of the Bunch Filling information in the LHCIF part of GRP. I need to wait for  https://github.com/AliceO2Group/AliceO2/pull/7367/files to be merged, then I can fix the (anti)clockwise) beam indices. 
Last to be implemented is the workflow for filling the B field information. Then I think we have everything - to be reviewed, of course.
Chiara

**Comment by chiarazampolli** [comment]
Hi @shahor02 , 

In this PR I added the change for the path of the MeanVertex.
BTW, what if I remove the [WIP] from this PR, and we use another one for the processing of the B field Data Points?

Chiara

**Comment by shahor02** [comment]
@chiarazampolli thanks! BTW, the MeanVertexObject (and calibrator) should be eventually changed to account for the diamond slopes.
OK for removing WIP and eventually merging.

**Review by shahor02 — APPROVED** [review]
failures are unrelated.

**Pull request merged**